app/models/plan.rb

```diff
@@ -566,6 +570,13 @@ def landing_page
     identifiers.select { |i| %w[doi ark].include?(i.identifier_format) }.first
   end
 
+  # Returns the identifier associated with the grant_id
+  def grant
+    return nil unless grant_id.present?
+
+    identifiers.select { |identifier| identifier.id == grant_id }.first
+  end
```

Comment:
I'd like to handle this through Rails associations if we can.  would something like the following work?
```
has_one :grant, class_name: 'Identifier'
```
The id on the identifier should still be unique to that table, so it should be enough to build-out that reference.

Reply:
agreed. will give it a try

Reply:
The following seems to work:
`has_one :grant, as: :identifiable, dependent: :destroy, class_name: "Identifier"`
I did not need to update the schema migration. 

I also added some sanity checks in the `models/plan_spec.rb` to ensure things work properly